### PR TITLE
Add aegii

### PR DESCRIPTION
**Pull request by Spandamn**

### CODE REVIEW

File: data/mods/ssb/abilities.ts [hunk 1]
```diff
@@ -67,6 +67,37 @@ export const Abilities: {[k: string]: ModdedAbilityData} = {
...
+			let success = false;
+			for (const ally of side.pokemon) {
+				if (ally !== pokemon && ally.hasAbility('soundproof')) continue;
+				if (ally.cureStatus()) success = true;
+			}
+			if (pokemon.side.foe.active[0]) {
+				pokemon.side.foe.active[0].addVolatile('embargo');
+				this.add('-start', pokemon, 'Embargo');
+			}
+		},
```
Comment (ghost) [inline data/mods/ssb/abilities.ts]: honestly i'd rather if you use all 3 moves simultaneously a la Zodiax

File: data/mods/ssb/conditions.ts [hunk 2]
```diff
@@ -63,6 +63,50 @@ export const Conditions: {[k: string]: ModdedConditionData} = {
...
+					move: moveData.name,
+					id: moveData.id,
+					pp: ((moveData.noPPBoosts || moveData.isZ) ? moveData.pp : moveData.pp * 8 / 5),
+					maxpp: ((moveData.noPPBoosts || moveData.isZ) ? moveData.pp : moveData.pp * 8 / 5),
+					target: moveData.target,
+					disabled: false,
+					disabledSource: '',
+					used: false,
+				});
+			}
```
Comment (ghost) [inline data/mods/ssb/conditions.ts]: There is a place in random-teams for moveset modifications, if you need help lmk

File: data/mods/ssb/conditions.ts [hunk 3]
```diff
@@ -63,6 +63,50 @@ export const Conditions: {[k: string]: ModdedConditionData} = {
...
+	aegii: {
+		noCopy: true,
+		onStart(pokemon) {
+			let quotes: string[] = [
+				`stream fiesta!!! https://youtu.be/eDEFolvLn0A`,
+				`stream more&more!!! https://youtu.be/mH0_XpSHkZo`,
+				`stream wannabe!!! https://youtu.be/fE2h3lGlOsk`,
+				`stream love bomb!!! https://youtu.be/-SK6cvkK4c0`
+			];
+			this.add(`c|${getName('aegii')}|${quotes[this.random(4)]}`);
```
Comment (ghost) [inline data/mods/ssb/conditions.ts]:
```suggestion
			this.add(`c|${getName('aegii')}|${[`stream fiesta!!! https://youtu.be/eDEFolvLn0A`, `stream more&more!!! https://youtu.be/mH0_XpSHkZo`, `stream wannabe!!! https://youtu.be/fE2h3lGlOsk`,  `stream love bomb!!! https://youtu.be/-SK6cvkK4c0`][this.random(4)]}`);
```

File: data/mods/ssb/conditions.ts [hunk 4]
```diff
@@ -63,6 +63,50 @@ export const Conditions: {[k: string]: ModdedConditionData} = {
...
+			}
+		},
+		onSwitchOut() {
+			let quotes: string[] = [
+				`brb, buying albums`,
+				`brb, downloading fancams`,
+				`brb, streaming mvs`,
+				`brb, learning choreos`
+			];
+			this.add(`c|${getName('aegii')}|${quotes[this.random(4)]}`);
```
Comment (ghost) [inline data/mods/ssb/conditions.ts]:
```suggestion
			this.add(`c|${getName('aegii')}|${[`brb, buying albums`, `brb, buying albums`, `brb, streaming mvs`, `brb, learning choreos`][this.random(4)]}`);
```

File: data/mods/ssb/abilities.ts (hunk 1 above)
Comment (Spandamn) [reply data/mods/ssb/abilities.ts]: I feel going this way feels better cause then it feels like an ability instead of simply using the moves

Comment (ghost) [reply data/mods/ssb/abilities.ts]: alright fair enough. That looks good

File: data/mods/ssb/random-teams.ts [hunk 5]
```diff
@@ -504,6 +510,9 @@ export class RandomStaffBrosTeams extends RandomTeams {
 
 			// Any set specific tweaks occur here.
 			if (set.name === 'quadrophenic') set.moves[this.random(2) + 1] = 'Conversion';
+			if (set.name === 'aegii' && this.random(2) === 1) {
+				set.moves = ['Shadow Ball', 'Flash Cannon', set.moves[2], set.moves[4]];
+			}
```
Comment (ghost) [inline data/mods/ssb/random-teams.ts]:
```suggestion
			if (set.name === 'aegii' && this.randomChance(1, 2)) {
				set.moves[set.moves.indexOf('Shadow Claw')] = 'Shadow Ball';
				set.moves[set.moves.indexOf('Iron Head')] = 'Flash Cannon';
			}
```

**Comment by Spandamn** [comment]
@xInstruct done it wont test

File: data/mods/ssb/moves.ts [hunk 6]
```diff
@@ -97,6 +97,76 @@ export const Moves: {[k: string]: ModdedMoveData} = {
...
+					this.boost({atk: -2}, source, target, this.dex.getActiveMove("K-Shield"));
+				}
+				const boost: {[key: string]: number} = {};
+				if (source.m.hasMovesetFixed) {
+					boost['spa'] = 1;
+				} else {
+					boost['atk'] = 1;
+				}
+				this.boost(boost, source);
+				return this.NOT_FAIL;
```
Comment (ghost) [inline data/mods/ssb/moves.ts:173]: ????

File: data/mods/ssb/moves.ts [hunk 7]
```diff
@@ -97,6 +97,76 @@ export const Moves: {[k: string]: ModdedMoveData} = {
...
+				} else {
+					this.boost({atk: -2}, source, target, this.dex.getActiveMove("K-Shield"));
+				}
+				const boost: {[key: string]: number} = {};
+				if (source.m.hasMovesetFixed) {
+					boost['spa'] = 1;
+				} else {
+					boost['atk'] = 1;
+				}
+				this.boost(boost, source);
```
Comment (ghost) [inline data/mods/ssb/moves.ts]: to make this more mon friendly, just cycle through all the moves and run 2 counters. Boost the relative stat depending on whether which is more

File: data/mods/ssb/conditions.ts [hunk 8]
```diff
@@ -63,6 +63,21 @@ export const Conditions: {[k: string]: ModdedConditionData} = {
 			this.add(`c|${getName('Aelita')}|X.A.N.A. is finally finished for good.`);
 		},
 	},
+	aegii: {
+		noCopy: true,
+		onStart(pokemon) {
+			this.add(`c|${getName('aegii')}|${[`stream fiesta!!! https://youtu.be/eDEFolvLn0A`, `stream more&more!!! https://youtu.be/mH0_XpSHkZo`, `stream wannabe!!! https://youtu.be/fE2h3lGlOsk`, `stream love bomb!!! https://youtu.be/-SK6cvkK4c0`][this.random(4)]}`);
+			// Decide whether aegii will be special or not
+			if (!pokemon.m.hasMovesetFixed) return;
+			if (pokemon.set.moves[0] === "Shadow Ball") pokemon.m.hasMovesetFixed = true; // aegii has a special moveset.
```
Comment (ghost) [inline data/mods/ssb/conditions.ts]: see previous comment, you wont need these after above

Comment (Spandamn) [reply data/mods/ssb/conditions.ts]: I think this approach is better since all you have to do is just set a flag instead of runninga counter everytime

Comment (ghost) [inline data/mods/ssb/conditions.ts]:
```suggestion
```

File: data/mods/ssb/moves.ts [hunk 9]
```diff
@@ -97,6 +97,87 @@ export const Moves: {[k: string]: ModdedMoveData} = {
...
+				}
+				let specCount = 0;
+				let physCount = 0;
+				for (const moveSlot of target.moveSlots) {
+					const moveid = moveSlot.id;
+					const theMove = this.dex.getMove(moveid);
+					if (theMove.category === "Special") {
+						specCount++;
+					} else if (theMove.category === "Physical") {
+						physCount++
```
Comment (ghost) [inline data/mods/ssb/moves.ts]:
```suggestion
						physCount++;
```

**Pull request merged**